ttnn/cpp/ttnn/device_operation.hpp

```diff
@@ -332,6 +340,12 @@ void launch_on_worker_thread(auto cq_id, auto operation_id, const auto& operatio
             },
             program);
 
+            for (auto& cb : program_ptr->circular_buffers()) {
+                tt::tt_metal::GraphTracker::instance().track_allocate_cb(cb->core_ranges(), 0, cb->size());
```

Comment:
Are we looping through the buffers every time whether we turn off this feature or not?  This looks like a possible perf hit to me.

Reply:
I am going to change this code. But it is not a big deal we have like 2-4 buffers at most.

Reply:
This comment is incorrect. We have 32 available cbs that can be used and there are some ops that use a lot more than 4

Reply:
put it under the additional check in the new function track_program. So overhead will be just +1 if.